File: Remove Duplicates - GFG/remove-duplicates.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function template for C++
class Solution{
public:	
		
	string removeDups(string S) 
	{
	    // Your code goes here
	    for(int i=0 ; i<S.size()-1 ; i++){
	        if(S[i]=='_')
	            continue;
	        for(int j=i+1 ; j<S.size() ; j++){
	            if(S[i]==S[j]){
	                S[j]='_';
	            }
	        }
	    }
	    string res="";
	    for(char i: S){
	        if(i!='_'){
	            res += i;
	        }
	    }
	    return res;
	}
};

//{ Driver Code Starts.


int main() 
{
   	

   	ios_base::sync_with_stdio(0);
    cin.tie(NULL);
    cout.tie(NULL);
   
   	int t;
   	cin >> t;
   	while(t--)
   	{
   		string s;
   		cin >> s;

   		
   		Solution ob;
   		cout << ob.removeDups(s) << "\n";
   		
   	}

    return 0;
}
// } Driver Code Ends

File: Remove Duplicates - GFG/remove-duplicates.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String s = read.readLine();
            
            Solution ob = new Solution();
            String result = ob.removeDups(s);
            
            System.out.println(result);
        }
    }
}
// } Driver Code Ends


//User function Template for Java

class Solution {
    String removeDups(String S) {
        // code here
        int alpha[] = new int[26];
        String res = "";
        for(int i = 0 ; i < S.length() ; i++){
            char ch = S.charAt(i);
            int d = ch-'a';
            if(alpha[d] == 0){
                res += ch;
            }
            alpha[d]++;
        }
        return res;
    }
}